internal/buildengine/languageplugin/go_plugin_helpers.go

```diff
@@ -0,0 +1,123 @@
+package languageplugin
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/TBD54566975/ftl/internal/errors"
+	"golang.org/x/mod/modfile"
+)
+
+func replacementWatches(moduleDir, deployDir string) ([]string, error) {
```

Comment:
all this code was previously in `moduleconfig` but is all specific to the go plugin